### PR TITLE
Make usage of Ankerl map more widespread

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- The Ankerl maps were not used optimally everywhere where they could be

# Description of the solution

- Add this. Also took the opportunity to optimize one part of fine search

# How Has This Been Tested?

- All unit tests and examples

# Comments

- This did not change the performance significantly. I will need to profile the code again to identify bottlenecks now.

### CODE REVIEW

File: source/dem/particle_particle_fine_search.cc [hunk 1]
```diff
@@ -23,22 +23,25 @@ ParticleParticleFineSearch<dim>::particle_particle_fine_search(
   for (auto &&adjacent_particles_list :
        adjacent_particles | boost::adaptors::map_values)
     {
+      if (adjacent_particles_list.empty())
+        continue;
+
+      // Gather information about particle 1
+      auto &particle_one = adjacent_particles_list.begin()->second.particle_one;
+      Point<dim, double> particle_one_location = particle_one->get_location();
+
```
Comment (acdaigneault) [inline source/dem/particle_particle_fine_search.cc:32]: Nice catch

File: source/dem/particle_particle_fine_search.cc [hunk 2]
```diff
@@ -23,22 +23,25 @@ ParticleParticleFineSearch<dim>::particle_particle_fine_search(
...
       // Iterating over each map which contains the contact information
-      // including particles I and II
       for (auto adjacent_particles_list_iterator =
              adjacent_particles_list.begin();
            adjacent_particles_list_iterator != adjacent_particles_list.end();)
         {
-          // Getting contact information and particles I and II as local
+          // Getting contact information and particle II as local
```
Comment (acdaigneault) [inline source/dem/particle_particle_fine_search.cc:38]:
```suggestion
          // Getting contact information and particle 2 as local
```

**Review by acdaigneault — APPROVED** [review]
No update of the performance analysis tests :( ?

**Comment by blaisb** [comment]
> No update of the performance analysis tests :( ?

It did not change significantly enough :). I'll remake one of master this week.

File: source/dem/particle_particle_fine_search.cc (hunk 1 above)
Comment (blaisb) [reply source/dem/particle_particle_fine_search.cc:32]: Thanks :)

**Pull request merged**